### PR TITLE
adapt Customized kout/kneighbor, template path and  jaccard similar

### PR DESCRIPTION
**Pull request by zhoney**

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/structure/traverser/RepeatEdgeStep.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+import com.baidu.hugegraph.api.API;
+import com.baidu.hugegraph.api.traverser.TraversersAPI;
+import com.baidu.hugegraph.structure.constant.Direction;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class RepeatEdgeStep extends EdgeStep {
+
+    @JsonProperty("max_times")
+    public int maxTimes = 1;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/traverser/RepeatEdgeStep.java:33]: also add minTimes at this time

File: src/main/java/com/baidu/hugegraph/structure/traverser/VerticesArgs.java [hunk 2]
```diff
@@ -78,13 +78,13 @@ public Builder property(String key, Object value) {
...
 
-        protected SourceVertices build() {
+        protected VerticesArgs build() {
             E.checkArgument(!((this.sources.ids == null ||
                                this.sources.ids.isEmpty()) &&
                               (this.sources.properties == null ||
                                this.sources.properties.isEmpty()) &&
                               this.sources.label == null),
-                            "No source vertices provided");
+                            "No vertices provided");
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/traverser/VerticesArgs.java:87]: improve the check logic

File: src/main/java/com/baidu/hugegraph/driver/TraverserManager.java [hunk 3]
```diff
@@ -139,6 +149,11 @@ public double jaccardSimilarity(Object vertexId, Object otherId,
                                              label, degree);
     }
 
+    public Map<Object, Double> jaccardSimilarity(
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/driver/TraverserManager.java:152]: can the two jaccardSimilarity methods be distinguished?

File: src/main/java/com/baidu/hugegraph/structure/traverser/RepeatEdgeStep.java (hunk 1 above)
Comment (javeme) [reply src/main/java/com/baidu/hugegraph/structure/traverser/RepeatEdgeStep.java:33]: also add minTimes at this time

File: src/main/java/com/baidu/hugegraph/structure/traverser/RepeatEdgeStep.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    public static class Builder {
+
+        protected RepeatEdgeStep step;
+
+        public Builder() {
+            this.step = new RepeatEdgeStep();
+        }
+
+        public Builder direction(Direction direction) {
+            this.step.direction = direction.toString();
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/structure/traverser/RepeatEdgeStep.java:58]: change type of step.direction to Direction

File: src/test/java/com/baidu/hugegraph/api/traverser/CustomizedPathsApiTest.java [hunk 5]
```diff
@@ -109,16 +109,16 @@ public void testCustomizedPathsSourceIds() {
         Object lopId = getVertexId("software", "name", "lop");
         Object peterId = getVertexId("person", "name", "peter");
 
-        PathsRequest.Builder builder = new PathsRequest.Builder();
+        CustomizedPathsRequest.Builder builder = new CustomizedPathsRequest.Builder();
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/api/traverser/CustomizedPathsApiTest.java:112]: CustomizedPathsRequest.builder()

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**